src/main/java/Deadline.java

```diff
@@ -0,0 +1,53 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * @author bakwxh
+ * @version 0.1
+ */
+public class Deadline extends Task {
+	/**
+	 * By time of deadline.
+	 */
+	private Date by;
+
+    /**
+     * Constructor.
+     * @param description Description.
+     * @param by By time.
+     */
+    public Deadline(final String description, final Date by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Shows task.
+     */
+    @Override
+    public String showTask() {
+        return "[D]" + super.showTask() + " (by: " + by.toString() + ")";
```

Comment:
A good way of using super.showTask()! 
I was wondering how to minimise the repetitions required in displaying the done status and the task name. 
This could be one way!